react/features/recent-list/components/RecentList.native.js

```diff
@@ -1,202 +1,241 @@
-import React from 'react';
-import { ListView, Text, TouchableHighlight, View } from 'react-native';
+// @flow
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
 
-import { Icon } from '../../base/font-icons';
-
-import AbstractRecentList, { _mapStateToProps } from './AbstractRecentList';
-import { getRecentRooms } from '../functions';
-import styles, { UNDERLAY_COLOR } from './styles';
+import { appNavigate } from '../../app';
+import {
+    getLocalizedDateFormatter,
+    getLocalizedDurationFormatter,
+    translate
+} from '../../base/i18n';
+import { NavigateSectionList } from '../../base/react';
+import { parseURIString } from '../../base/util';
 
 /**
- * The native container rendering the list of the recently joined rooms.
- *
- * @extends AbstractRecentList
+ * The type of the React {@code Component} props of {@link RecentList}
  */
-class RecentList extends AbstractRecentList {
+type Props = {
+
     /**
-     * The datasource wrapper to be used for the display.
+     * Renders the list disabled.
      */
-    dataSource = new ListView.DataSource({
-        rowHasChanged: (r1, r2) =>
-            r1.conference !== r2.conference
-                && r1.dateTimeStamp !== r2.dateTimeStamp
-    });
+    disabled: boolean,
 
+    /**
+     * The redux store's {@code dispatch} function.
+     */
+    dispatch: Dispatch<*>,
+
+    /**
+     * The translate function.
+     */
+    t: Function,
+
+    /**
+     * The default server URL.
+     */
+    _homeServer: string,
```

Comment:
I don't know if I like the name "_homeServer" only because I usually see xxxURL used instead. Why not call is defaultServerURL or something along those lines?